code/modules/martial_arts/muscle_implant.dm

```diff
@@ -0,0 +1,55 @@
+/datum/martial_art/muscle_implant
+	name = "Empowered Muscle Implant Fighting"
+	weight = 1
+	/// Are we EMP'ed?
+	var/is_emp
+	/// How much damage should our punches deal
+	var/punch_damage = 13
+	var/picked_hit_type
+
+/datum/martial_art/muscle_implant/harm_act(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	MARTIAL_ARTS_ACT_CHECK
+
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		to_chat(user, "<span class='warning'>You don't want to hurt [target]!</span>")
+		return FALSE
+	picked_hit_type = pick("punch", "smash", "kick")
+	if(ishuman(target))
+		if(target.check_shields(user, punch_damage, "[user]'s' [picked_hit_type]"))
+			user.do_attack_animation(target)
+			playsound(target.loc, 'sound/weapons/punchmiss.ogg', 25, TRUE, -1)
+			return FALSE
+	if(is_emp)
+		do_the_punch(user, user)
+	else
+		do_the_punch(user, target)
+	playsound(target.loc, 'sound/weapons/resonator_blast.ogg', 50, TRUE)
+	playsound(target.loc, 'sound/weapons/genhit2.ogg', 50, TRUE)
+
+	return TRUE
+
+/datum/martial_art/muscle_implant/proc/do_the_punch(mob/living/carbon/human/puncher, mob/living/carbon/human/punchee)
+	puncher.do_attack_animation(punchee, ATTACK_EFFECT_SMASH)
+	add_attack_logs(puncher, punchee, "Melee attacked with martial-art [src] : Punched", ATKLOG_ALL)
+	punchee.apply_damage(punch_damage, BRUTE, puncher.zone_selected)
+
+	if(!IS_HORIZONTAL(puncher)) //Throw them if we are standing
+		var/atom/throw_target = get_edge_target_turf(punchee, puncher.dir)
+		punchee.throw_at(throw_target, rand(1, 2), 0.5, puncher)
+
+	punchee.visible_message(
+		"<span class='danger'>[puncher] [picked_hit_type]ed [punchee]!</span>",
+		"<span class='danger'>You're [picked_hit_type]ed by [puncher]!</span>",
+		"<span class='danger'>You hear user sickening sound of flesh hitting flesh!</span>",
+	)
+
+	to_chat(puncher, "<span class='danger'>You [picked_hit_type] [punchee]!</span>")
+
+/datum/martial_art/muscle_implant/proc/emp_act(severity, mob/owner)
+	is_emp = TRUE
```

Comment:
This is just a nitpick on my part, but is_emp should probably be in the past tense whether that be `emped` or `is_emped` or another variant, I think it would be better